code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Offer.java

```diff
@@ -472,16 +473,16 @@ static Offer fromEventData(final Map<String, Object> data) {
                 }
 
                 final String format =
-                        (String) offerData.get(OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_FORMAT);
-                if (OptimizeUtils.isNullOrEmpty(format)) {
-                    Log.debug(
-                            OptimizeConstants.LOG_TAG,
-                            SELF_TAG,
-                            "Cannot create Offer object, provided data Map doesn't contain valid"
-                                    + " item data format.");
-                    return null;
-                }
-
+                        DataReader.getString(
+                                offerData, OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_FORMAT);
+                final OfferType offerType =
+                        (format != null)
+                                ? OfferType.from(format)
+                                : OfferType.from(
+                                        DataReader.optString(
```

Comment:
Can we just do `OfferType.from(DataReader.getString(offerData, OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_TYPE))`? 
`DataReader.getString` would return `null` if the`type` is not found. `OfferType.from` already has a check to return `UNKNOWN` if the input is null. 
`DataReader.getString` would throw an exception if `type` is not a string, which is an error case

Reply:
Mayeb I was a bit unclear in my comment here. I meant to convey something to this effect
```
final String format =
                        DataReader.getString(
                                offerData, OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_FORMAT);
                final OfferType offerType =
                        (format != null)
                                ? OfferType.from(format)
                                : OfferType.from(
                                        DataReader.getString(offerData, OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_TYPE
                                        ));
```

Reply:
done @spoorthipujariadobe